webapp/src/components/ImageFieldEditor.tsx

```diff
@@ -42,16 +45,22 @@ export class ImageFieldEditor<U extends pxt.Asset> extends React.Component<Image
 
     constructor(props: ImageFieldEditorProps) {
         super(props);
+        this.toggleFilter = this.toggleFilter.bind(this);
```

Comment:
I was confused why I needed to do the binding here but the other functions didn't? I feel like I'm missing somethin obvious >>

Reply:
You need to do it for ones that are being passed as parameters to react components (e.g. in this case it's acting as a handler for an event, so you need to do a bind or define it as a field (e.g. instead of `toggleFilter() {` defining it as `toggleFilter = () => {`) Just a perf thing, if I recall correctly it's to make it compare as equal when react checks props / to prevent unnecessary renders

Reply:
ooooo gotcha thanks!

Reply:
do we have a preference for binding vs field?

Reply:
my preference is towards arrow function defined as a property over `bind` / I think that's common to other people on the team, but it doesn't make a big difference at all